src/widgets/frigate/proxy.js

```diff
@@ -0,0 +1,139 @@
+import getServiceWidget from "utils/config/service-helpers";
+import createLogger from "utils/logger";
+import { formatApiCall, sanitizeErrorURL } from "utils/proxy/api-helpers";
+import { addCookieToJar, setCookieHeader } from "utils/proxy/cookie-jar";
+import { httpProxy } from "utils/proxy/http";
+import validateWidgetData from "utils/proxy/validate-widget-data";
+import widgets from "widgets/widgets";
+
+const proxyName = "frigateProxyHandler";
+const logger = createLogger(proxyName);
+
+export default async function frigateProxyHandler(req, res, map) {
+  const { group, service, endpoint, index, url } = req.query;
+
+  if (group && service) {
+    const widget = await getServiceWidget(group, service, index);
+
+    if (!widgets?.[widget.type]?.api) {
+      return res.status(403).json({ error: "Service does not support API calls" });
+    }
+
+    if (widget) {
+      // If there are more than one question marks, replace others to &
+      let urlString = formatApiCall(widgets[widget.type].api, { endpoint, ...widget }).replace(/(?<=\?.*)\?/g, "&");
+      if (widget.type === "customapi" && widget.url?.endsWith("/")) {
+        urlString += "/"; // Ensure we dont lose the trailing slash for custom API calls
+      }
+      const url = new URL(urlString);
+      const loginUrl = `${widget.url}/api/login`;
```

Comment:
```suggestion
      const url = formatApiCall(widgets[widget.type].api, { endpoint, ...widget });
      const loginUrl = `${widget.url}/api/login`;
```

Yea, this proxy will only be used by frigate, so `widget.type` should always be `frigate` (and not `customapi`)